src/views/VideoView/VideoView.tsx

```diff
@@ -40,15 +40,19 @@ const VideoView: React.FC<RouteComponentProps> = () => {
       return
     }
     const currentVideo = state.watchedVideos.find((v) => v.id === data?.video?.id)
-    const duration = data?.video?.duration ? Number(data?.video?.duration) : 0
 
-    if (timestampFromQuery && timestampFromQuery < duration) {
-      setStartTimestamp(timestampFromQuery)
+    setStartTimestamp(currentVideo?.__typename === 'INTERRUPTED' ? currentVideo.timestamp : 0)
+  }, [data?.video?.id, state.watchedVideos, startTimestamp, data?.video?.duration])
+
+  useEffect(() => {
+    const duration = data?.video?.duration ?? 0
+    if (!timestampFromQuery || timestampFromQuery > duration) {
+      return
     }
-    if (!timestampFromQuery) {
-      setStartTimestamp(currentVideo?.__typename === 'INTERRUPTED' ? currentVideo.timestamp : 0)
+    if (timestampFromQuery < duration) {
```

Comment:
I don't think we need that `if`, right? Since this is already taken care of above